### ISSUE TITLE
System.ArgumentOutOfRangeException with nested list of objects

### ISSUE THREAD

**Issue by JMRodriguez24** [msgpack/msgpack-cli#63]
Here is a gist with a failing test:

https://gist.github.com/JMRodriguez24/32c3892c6223c90f255f#file-msgpack_cli_exception

Here is the exception:

```System.ArgumentOutOfRangeException : 'count' is negative.
Parameter name: count
   at MsgPack.Packer.PackArrayHeaderCore(Int32 count) in Packer.cs: line 992
   at MsgPack.Packer.PackArrayHeader(Int32 count) in Packer.cs: line 973
   at MsgPack.Serialization.EmittingSerializers.Generated.Samples_PTaxHistSerializer2.PackToCore(Packer, PTaxHist)
   at MsgPack.Serialization.MessagePackSerializer`1.PackTo(Packer packer, T objectTree) in MessagePackSerializer`1.cs: line 225
   at MsgPack.Serialization.EmittingSerializers.Generated.System_Collections_Generic_IList_1_Samples_PTaxHist_Serializer1.PackToCore(Packer, IList`1)
   at MsgPack.Serialization.MessagePackSerializer`1.PackTo(Packer packer, T objectTree) in MessagePackSerializer`1.cs: line 225
   at MsgPack.Serialization.EmittingSerializers.Generated.Samples_PayRegSerializer0.PackToCore(Packer, PayReg)
   at MsgPack.Serialization.MessagePackSerializer`1.PackTo(Packer packer, T objectTree) in MessagePackSerializer`1.cs: line 225
   at MsgPack.Serialization.MessagePackSerializer`1.Pack(Stream stream, T objectTree) in MessagePackSerializer`1.cs: line 173
   at Samples.BasicUsageSample.SerializeThenDeserialize() in Sample01_BasicUsage.cs: line 32```

**Comment by yfakariya** [comment]
Sorry, it is a bug related to IL generation. It occurs when serializing member count is between 128 and 255. I'll fix in next release.

**Comment by JMRodriguez24** [comment]
Great!  Thanks!

**Comment by yfakariya** [comment]
This issue was fixed in 0.5.9.
